File: benchmarks/runtime/catch_overhead.js
let n = 500000;
let i = 0;
let sum = 0;
while (i < n) {
  try {
    sum += 1;
  } catch (e) {
    sum -= 1;
  }
  i += 1;
}
if (sum <= 0) process.exit(1);
console.log(sum);

File: benchmarks/runtime/catch_overhead.lua
local n = 500000
local i = 0
local sum = 0
while i < n do
  local ok = pcall(function()
    sum = sum + 1
  end)
  if not ok then
    sum = sum - 1
  end
  i = i + 1
end
if sum <= 0 then os.exit(1) end
print(sum)
